goon/browserassets/js/browserOutput.js

```diff
@@ -157,13 +158,13 @@ function setHighlightColor(match) {
 
 //Highlights words based on user settings
 function highlightTerms(el) {
-	var element = $(el)
-	if(!(element.mark)) { // mark.js isn't loaded; give up
-		return
-	}
 	for (var i = 0; i < opts.highlightTerms.length; i++) { //Each highlight term
 		if(opts.highlightTerms[i]) {
-			element.mark(opts.highlightTerms[i], {"element" : "span", "each" : setHighlightColor});
+			var rexp = new RegExp("("+opts.highlightTerms[i]+")","gmi")
+			if(!opts.highlightRegexEnable){
+				rexp = new RegExp("("+opts.highlightTerms[i].replace(new RegExp("([^a-zA-Z0-9])","gmi"),"\\$1")+")","gmi")
+			}
+			el.innerHTML = el.innerHTML.replace(rexp,"<span style=\"background-color:"+opts.highlightColor+"\">$1</span>")
```

Comment:
Should this be `innerText` and not `innerHTML`? Does this break html tags if you set the regex to something like "span" or "warning"?

Reply:
Thats true, however if you're using regex you should understand that its to be expected if you dont write your regex correctly, ie: if you want to highlight span or warning without fucking the html, youd \b(span|warning)\b , where \b is the token for word boundary.

Reply:
That doesn't sound like the typical user case. Most users don't have an understanding of the HTML that goes into the chat window, and seeing as there's no way to view the HTML content of the chat window without code diving, even the users that would want to change it probably wouldn't bother. Most users just want to highlight "AI" when "AI" is spoken, and will be confused when something like that breaks their window.

Also, this basically gives the user the ability to insert any html they want into their chat window. While that's not really a genuine security risk, Byond right now as far as I know doesn't let you do this. My feeling is that there's no need to edit the html so why complicate things? Not to mention that trying to parse html with regex is not a great idea: https://stackoverflow.com/questions/6751105/why-its-not-possible-to-use-regex-to-parse-html-xml-a-formal-explanation-in-la

Have you tried this with `innerText` instead? I'm almost certain that would be a better user experience for the vast majority of our users.

Reply:
This wont break anyone's previously defined highlights, Regex is an option here that is effectively disabled by default, And no, you cannot insert any html anywhere, as this is a replace with a predefined replacement, therefore not making it an issue at all. The point is less of parsing the html, and more of parsing around while preserving the html.

Reply:
> And no, you cannot insert any html anywhere, as this is a replace with a predefined replacement

You're right, I forgot about that.

I still don't see any use case for trying to read the HTML itself.

Reply:
You can use noncapturing groups to anchor to a specific location before reading ahead when you want to highlight something, but its dynamic and unknown, which is where reading the html would come in handy.

Reply:
Right now highlighting "warning" would break all the warning classes since `<span class="warning">the mouse nuzzles you</span>`` becomes `<span class="<span style="background-color:yellow">warning</span>">the mouse nuzzles you</span>`.
In other words, would break spectacularly.

So instead of matching HTML you should really walk over DOM tree anchored on the `el` and and only mess with text(!) nodes by wrapping them in the colored span.

Separately, if the eventual goal of this PR is to highlight all warnings - you be to add custom CSS that would augment the `warning` class. (That custom CSS supportmight already exist? I'm unsure.. How does dark chat theme work? )

Reply:
Totally forgot i could just match on innerText when the whole regex bit is off and innerHtml when its on. I brain.

Reply:
You would still have the issue of people searching for `\bwarning\b` (or, hell, `war`)in regex mode would have their HTML messed up.

I strongly suggest you don't match on HTML, in either mode.

Reply:
Limiting to the text only severely limits the flexibility goal of the system in the first place.